scripts/lint-lists.py

```diff
@@ -115,10 +166,11 @@ def main(source='OONI', notes='', legacy=False, fix_duplicates=False):
                         InvalidURL(url, csv_path, idx+2)
                     )
                 url = url.strip().lower()
-                canonical_url = url
-                if url.endswith('/'):
-                    # We strip trailing / for canonical URLs
-                    canonical_url = url[:-1]
+                canonical_url = get_canonical_url(url)
+                if canonical_url != url and canonical_check:
+                    errors.append(
+                        NonCanonicalURL(url + ' instead of ' + canonical_url, csv_path, idx+2)
```

Comment:
It's interesting that you put `csv_path` as separate argument, but format `url` and `canonical_url` into a message string.

Reply:
That is done because it's common to all the exceptions. If you look at the structure of the exception handlers, the format is `Exception(human_meaningful_thing, csv_path, csv_file_row)`.

Maybe the cleanest thing would be to create a new exception type for this particular sort of exception, but this seemed good enough.